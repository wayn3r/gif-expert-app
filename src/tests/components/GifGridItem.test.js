import { shallow } from 'enzyme'
import GifGridItem from '../../components/GifGridItem'

describe('Pruebas para <GifGridItem />', () => {
    const title = 'Titulo del GIF'
    const url = 'https://localhost/algo.jpg'
    const wrapper = shallow(<GifGridItem title={title} url={url} />)

    test('debe mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    })
    test('debe de tener un parrafo con el titulo', () => {
        const p = wrapper.find('p').text()
        expect(p).toBe(title)
    })
    test('debe de tener la imagen con url y alt de los props', () => {
        const img = wrapper.find('img').props()
        expect(img.src).toBe(url)
        expect(img.alt).toBe(title)
    })

    test('debe de tener la clase animate__animated', () => {
        const div = wrapper.find('div')
        // div.prop('className').includes('animate__animated')
        expect(div.hasClass('animate__animated')).toBe(true)
    })
})
