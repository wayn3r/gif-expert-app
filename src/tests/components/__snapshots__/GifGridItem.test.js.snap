// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para <GifGridItem /> debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulo del GIF"
    src="https://localhost/algo.jpg"
  />
  <p>
    Titulo del GIF
  </p>
</div>
`;
